pkg/asset/agent/gencrypto/authconfig.go

```diff
@@ -314,19 +323,28 @@ func GetAuthTokenFromCluster(ctx context.Context, kubeconfigPath string) (string
 	if err != nil {
 		return "", err
 	}
-	authToken, err := extractAuthTokenFromSecret(retrievedSecret)
+	_, _, watcherAuthToken, err := extractAuthTokensFromSecret(retrievedSecret)
 	if err != nil {
 		return "", err
 	}
-	return authToken, err
+	return watcherAuthToken, err
 }
 
-func extractAuthTokenFromSecret(secret *corev1.Secret) (string, error) {
-	existingWatcherAuthToken, exists := secret.Data[authTokenSecretDataKey]
+func extractAuthTokensFromSecret(secret *corev1.Secret) (string, string, string, error) {
+	existingAgentAuthToken, exists := secret.Data[userAuthKey]
+	if !exists || len(existingAgentAuthToken) == 0 {
+		return "", "", "", fmt.Errorf("auth token secret %s/%s does not contain the key %s or is empty", authTokenSecretNamespace, authTokenSecretName, agentAuthKey)
+	}
+
+	existingUserAuthToken, exists := secret.Data[userAuthKey]
+	if !exists || len(existingUserAuthToken) == 0 {
+		return "", "", "", fmt.Errorf("auth token secret %s/%s does not contain the key %s or is empty", authTokenSecretNamespace, authTokenSecretName, userAuthKey)
+	}
+	existingWatcherAuthToken, exists := secret.Data[watcherAuthKey]
 	if !exists || len(existingWatcherAuthToken) == 0 {
-		return "", fmt.Errorf("auth token secret %s/%s does not contain the key %s or is empty", authTokenSecretNamespace, authTokenSecretName, authTokenSecretDataKey)
+		return "", "", "", fmt.Errorf("auth token secret %s/%s does not contain the key %s or is empty", authTokenSecretNamespace, authTokenSecretName, watcherAuthKey)
```

Comment:
Note about the migration (4.17 -> 4.18): when executed the first time against a 4.17 cluster, the `createOrUpdateAuthTokenSecret` is going to fail due the missing data keys: ie, the secret is refreshed only when it does not exist or contains all the keys. I think it should be refreshed also when it exists and contains just the old key (meaning it's a 4.17 secret).

Reply:
nice finding 👌